src/realm/sync/binding_callback_thread_observer.cpp

```diff
@@ -19,5 +19,51 @@
 #include <realm/sync/binding_callback_thread_observer.hpp>
 
 namespace realm {
-BindingCallbackThreadObserver* g_binding_callback_thread_observer = nullptr;
+
+void BindingCallbackThreadObserver::call_did_create_thread(
+    const std::shared_ptr<BindingCallbackThreadObserver>& observer_ptr)
+{
+    // Call into the observer ptr if not null, otherwise, use the global thread observer
+    if (observer_ptr)
+        observer_ptr->did_create_thread();
+}
+
+void BindingCallbackThreadObserver::call_will_destroy_thread(
+    const std::shared_ptr<BindingCallbackThreadObserver>& observer_ptr)
+{
+    // Call into the observer ptr if not null, otherwise, use the global thread observer
```

Comment:
It looks like there is no global observer? Do we still need these static functions?

Reply:
These were just helper functions... I'll remove them and just use the direct function calls in the default socket provider.